expr/types/types.go

```diff
@@ -31,6 +31,20 @@ type MetricData struct {
 	AggregateFunction func([]float64, []bool) (float64, bool)
 }
 
+// New creates new MetricData with given metric timeseries. values have math.NaN() for absent
```

Comment:
looks like we don't have  math.NaN() for absent values, but keep a boolean array of absence.

Reply:
Indeed, updated comment!